design/design.md

```diff
@@ -15,6 +15,19 @@ After the user clicks on a list, the app shows all of the tasks for that list in
 
 More details about both of these views can be found in the section below.
 
+## Authentication
+Our app currently provides ways for users to login using their email and password, or using Google. Additionally, we have a sign up link which takes new users to a separate sign up page, and a password reset link which allows users to send a password reset email to themselves if they ever forget the password to the app.
+
+Because shared lists are accepted by default, we made it so that users must verify their emails before they are able to use the app.
+## Sharing Behavior
+On the main screen, lists owned by the user are shown first. This is intended behavior as we thought it would make more sense that most users would not want to group their personal lists with lists shared with them, even when following sorting rules. To differentiate between owned lists and shared lists, owned lists have the same 3 dots options menu, while shared lists have a person icon.
+Currently, owners of the list are able to expand the 3 dots out to show the "Share Lists" option, which opens up a popup that allows them to share the list with others. 
+For non list owners, they instead have the option the view the list sharing, which displays the owner of the list as well as all the other users the list is currently shared with.
+
+We thought that it wouldn't make sense for non-owners to leave a list that was shared with them that they do not want to be part of, as this would require them to be able to edit the shared permissions for that list, which implies that they would also be able to modify the permissions of other users. To circumvent this issue, we instead gave users the option to hide lists shared with them that they do not want to display instead. The user can then go into their main menu and unhide these shared lists at any time. 
```

Comment:
You could write a rule that ensures that the only difference between old and new shared permissions is that the user's email is removed (you wouldn't need to provide general editing permissions for that field).